Makefile

```diff
@@ -18,4 +18,6 @@ setup_db:
 	mix ecto.create
 	mix ecto.migrate
 
-setup_all: deps start_db setup_db
+setup_all: deps start_db
+	sleep 3
+	make setup_db
```

Comment:
Is the sleep 3  needed? Why?
Also remember that you don't need to add `make` inside a makefile. You can just do `setup_db`

Reply:
I added a 3 second sleep because if `setup_db` is run immediately after `start_db` (which starts the DB process), the DB may not be ready and listening for incoming connections (the container was started, but the DB process hasn't finished booting).

This causes `setup_db` which tries to connect to the DB to fail.

Reply:
It is updated here: [Update make call inside Makefile](https://github.com/CimimUxMaio/forth-evaluator/pull/10/commits/c9c2f0ecd5b54fbff82f75fbd1f2114a0cd39fd6)

Reply:
> but I found that the correct and safe way of calling another make command from inside another is by using the $(MAKE) variable.

But you're already using other make commands without using `$(MAKE)` in the snippet you mentioned:

```
setup_all: deps start_db
	sleep 3
	$(MAKE) setup_db
```

Here `deps` and `start_db` are makefile commands.

Reply:
Values next to the ":" (inlined) are not "instructions" but dependencies.

Make executes those before running "setup_all". I can not add setup_db to the dependency list since I need to call "sleep 3" first. `sleep 3` can not be added as a dependency since it is not a make command.